framework/src/dgkernels/DGLowerDKernel.C

```diff
@@ -41,15 +41,21 @@ DGLowerDKernel::DGLowerDKernel(const InputParameters & parameters)
     _test_lambda(_lowerd_var.phiLower())
 {
   const auto & lower_domains = _lowerd_var.activeSubdomains();
-  if (!lower_domains.count(Moose::INTERNAL_SIDE_LOWERD_ID) && lower_domains.size() != 1)
-    paramError(
-        "lowerd_variable",
-        "Must be only defined on the subdomain INTERNAL_SIDE_LOWERD_SUBDOMAIN subdomain that is "
-        "added by Mesh/build_all_side_lowerd_mesh=true");
-
-  if (_var.activeSubdomains().count(Moose::INTERNAL_SIDE_LOWERD_ID))
-    paramError("variable",
-               "Must not be defined on the subdomain INTERNAL_SIDE_LOWERD_SUBDOMAIN subdomain");
+  for (const auto & id : _mesh.interiorLowerDBlocks())
```

Comment:
if there is more code like these two checks that is shared between HFEM Kernels you should consider a base class

Reply:
`ArrayDGLowerDKernel` and `DGLowerDKernel` are the two base classes. Same for the two integrated BCs. `HFEMDirichletBC` needs one more check on a new coupled lower-d variable `u_hat`.